driver/src/test/scala/com/stratio/sparkta/driver/test/factory/CuratorFactoryHolderSpec.scala

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright (C) 2015 Stratio (http://stratio.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.stratio.sparkta.driver.test.factory
+
+import akka.event.slf4j.SLF4JLogging
+import com.stratio.sparkta.driver.factory.{SparkContextFactory, CuratorFactoryHolder}
+import com.typesafe.config.ConfigFactory
+import org.apache.curator.CuratorConnectionLossException
+import org.apache.curator.framework.api.ExistsBuilder
+import org.apache.curator.test.TestingServer
+import org.apache.zookeeper.CreateMode
+import org.junit.runner.RunWith
+import org.scalatest.junit.JUnitRunner
+import org.scalatest.{FlatSpec, _}
+
+/**
+ * This test specifies the behaviour of CuratorFactoryHolder that encapsulates the real curator's factory.
+ * @author anistal
+ */
+@RunWith(classOf[JUnitRunner])
+class CuratorFactoryHolderSpec extends FlatSpec with Matchers with BeforeAndAfter with GivenWhenThen with SLF4JLogging {
```

Comment:
Should't be this class an integration test? In that case it should be renamed to CuratorFactoryHolderIT